### PR TITLE
Onchain trackid via paymentReceiver & refundReceiver

### LINKED ISSUE #1056: [Onchain tracking] PoC
Create a PoC testing the changes for onchain data tracking. We will test the code for both final ways of adding the identifier.:

 - Append the identifier hex encoded (keccak256) at the end of the Data
 - Use the `refundReceiver` field to set the hex encoded identifier.

We are still waiting for feedback from the Protocol team to understand if the first approach may have security concerns. This needs to be confirmed after Devcon.

Coordinate with analytics team how they plan to gather the data

### PR DESCRIPTION
**Pull request by DaniSomoza**
## What it solves
Resolves #1056 

## How this PR fixes it

Added on chain tracking via [ `paymentReceiver` ](https://github.com/safe-global/safe-smart-account/blob/main/contracts/Safe.sol#L92) & [`refundReceiver`](https://github.com/safe-global/safe-smart-account/blob/main/contracts/Safe.sol#L120) fields

### Safe Deployment

Safe creation (v1.3.0)

Chain: Sepolia

TrackId: mi-test-track-id
Hashed trackId:  0xdda6d6dcc0edd564c5393808feb3b4e49d100cfe

Safe deployed v1.3.0 address: 0x00088ef66C3F981b07DB931f2302Ed1eB412E055

Deployment Link: https://sepolia.etherscan.io/tx/0xb0763692d198fb5df5546e3fa4bf3c431111c8aaf1010816924173c06f6e292d

Tenderly Link: https://dashboard.tenderly.co/tx/sepolia/0xb0763692d198fb5df5546e3fa4bf3c431111c8aaf1010816924173c06f6e292d/debugger?trace=0.1.1

Safe creation (v1.0.0)

Chain: gnosis

TrackId: mi-test-track-id
Hashed trackId:  0xdda6d6dcc0edd564c5393808feb3b4e49d100cfe

Safe deployed v1.0.0 address: 0x86e06Eb38B5f202a92417d06f20881B827d683CA

Deployment Link: https://gnosisscan.io/tx/0xc9803c5f2c38374d8705020ebf9600a4357c0d9e6a310dd04a3fd06d53dbdfb1

Tenderly Link: https://dashboard.tenderly.co/tx/gnosis-chain/0xc9803c5f2c38374d8705020ebf9600a4357c0d9e6a310dd04a3fd06d53dbdfb1/debugger?trace=0.1.0

### Safe Execution

Chain: Sepolia

TrackId: mi-test-track-id
Hashed trackId:  0xdda6d6dcc0edd564c5393808feb3b4e49d100cfe

Safe v1.3.0 address: 0x00088ef66C3F981b07DB931f2302Ed1eB412E055

Tenderly Link:  https://dashboard.tenderly.co/tx/sepolia/0x8d5871a6f668a3b4441f92d6f854c66dccc51ef53d3ae3362f7806c3011b1d4a/debugger?trace=0.1.3

transaction Hash: 0x8d5871a6f668a3b4441f92d6f854c66dccc51ef53d3ae3362f7806c3011b1d4a

### CODE REVIEW

**Comment by yagopv** [comment]
@DaniSomoza hey, i think you are adding the trackId to the safe transaction data right?

The trackId needs to be added to the Ethereum transaction data field because you cannot alter the information of the safe data field

cc: @akshay-ap

File: packages/protocol-kit/src/contracts/utils.ts [hunk 1]
```diff
@@ -84,6 +85,7 @@ export interface encodeSetupCallDataProps {
   customContracts?: ContractNetworkConfig
   customSafeVersion?: SafeVersion
   deploymentType?: DeploymentType
+  trackId: string
```
Comment (yagopv) [inline packages/protocol-kit/src/contracts/utils.ts:88]: should be optional?

Comment (DaniSomoza) [reply packages/protocol-kit/src/contracts/utils.ts]: done!

**Comment by DaniSomoza** [comment]
>  i think you are adding the trackId to the safe transaction data right?

No, we have 2 different POC to implement the on chain tracking. In this PR we are implementing the `trackId` in the `paymentReceiver` & `refundReceiver` fields.

[this is the PR](https://github.com/safe-global/safe-core-sdk/pull/1059) that adds the `trackId` to the safe transaction data field.

**Comment by DaniSomoza** [comment]
I'm closing this PR because it was intended as a proof of concept (PoC). After discussing with the team, we've decided to implement the alternative approach, appending the onchain identifier to the end of the data and calldata fields in transactions.

The final implementation can be found in this other PR: #1059

**Pull request closed without merging**